src/data_designer/config/utils/validation.py

```diff
@@ -288,6 +294,47 @@ def validate_drop_columns_processor(
     return []
 
 
+def validate_schema_transform_processor(
+    columns: list[ColumnConfigT],
+    processor_configs: list[ProcessorConfig],
+) -> list[Violation]:
+    violations = []
+
+    all_column_names = {c.name for c in columns}
+    for processor_config in processor_configs:
+        if processor_config.processor_type == ProcessorType.SCHEMA_TRANSFORM:
+            try:
+                json.dumps(processor_config.template)
+            except TypeError as e:
+                if "not JSON serializable" in str(e):
+                    violations.append(
+                        Violation(
+                            column=None,
+                            type=ViolationType.INVALID_TEMPLATE,
+                            message=f"Ancillary dataset processor {processor_config.name} template is not a valid JSON object.",
+                            level=ViolationLevel.ERROR,
+                        )
+                    )
```

Comment:
Sorry, we can do this in a follow-up PR, but I was thinking that we should have a pydantic validator for this. That way it will throw an error right away.